File: DIYHomeAutomation/app/src/main/java/com/example/diyhomeautomation/models/Device.kt
package com.example.diyhomeautomation.models

/**
 * Data class representing a device in the DIY Home Automation system.
 *
 * @property id The unique identifier for the device (nullable).
 * @property name The name of the device.
 * @property pinValue The pin value associated with the device (nullable).
 * @property state The state of the device (on/off) (nullable).
 * @property value The value associated with the device (nullable).
 * @property icon The icon representing the device (nullable).
 * @property roomId The ID of the room where the device is located.
 * @property typeDeviceId The ID of the device type (nullable).
 * @property taskDevices List of associated task devices (nullable).
 *
 * @constructor Creates an instance of Device with the provided properties.
 */
data class Device(
    var id: Int?,
    var name: String,
    var pinValue: String?,
    var state: Boolean?,
    var value: Double?,
    var icon: String?,
    var roomId: Int,
    var typeDeviceId: Int?,
    var taskDevices: List<TaskDevice>?
)

File: DIYHomeAutomationAPI/DIYHomeAutomationAPI/Models/Device.cs
using DIYHomeAutomationAPI.Models.Enums;

namespace DIYHomeAutomationAPI.Models
{
    /// <summary>
    /// This class represents a Device
    /// </summary>
    public class Device
    {
        /// <summary>
        /// Device's Id
        /// </summary>
        public int? Id { get; set; }

        /// <summary>
        /// Device's Name
        /// </summary>
        public string Name { get; set; } = null!;

        /// <summary>
        /// Device's PinValue
        /// </summary>
        public string? PinValue { get; set; } = null!;

        /// <summary>
        /// Device's State
        /// </summary>
        public bool? State { get; set; }

        /// <summary>
        /// Device's Value
        /// </summary>
        public double? Value { get; set; }        

        /// <summary>
        /// Device's Icon
        /// </summary>
        public string? Icon { get; set; }

        /// <summary>
        /// Room's Id
        /// </summary>
        public int? RoomId { get; set; }

        /// <summary>
        /// Device's Type Id
        /// </summary>
        public int? TypeDeviceId { get; set; }

        /// <summary>
        /// List of TaskDevice's
        /// </summary>
        public List<TaskDevice>? TaskDevices { get; set; }
    }
}
